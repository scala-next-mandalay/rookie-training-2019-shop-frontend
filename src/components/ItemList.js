import React from 'react';
import PropTypes from 'prop-types';
import { Box, Grid } from '@material-ui/core';
import Item from '../containers/Item';
import InfiniteScroll from 'react-infinite-scroller';
import ItemPropTypes from './ItemPropTypes';


const ItemList = ({ items, fetchItems, noMoreFetch }) => {
  return (
   
    <InfiniteScroll
        pageStart={0}
        loadMore={fetchItems}
        hasMore={!noMoreFetch}
        initialLoad={true}
        loader={<div className="loader" key={0}></div>}
    >
    
      <Box maxWidth={1300}  justify="center" mx="auto">
        <Grid container>{
          items.map(item => (
            <Grid key={item.id} item xs={12} sm={4} lg={4} md={4}>
              <Item key={item.id} row={item} />
            </Grid>
        
          ))
        }</Grid>
      </Box>
      
    
      
    </InfiniteScroll>
 
  );
};
ItemList.propTypes = {
  items: PropTypes.arrayOf(ItemPropTypes.isRequired).isRequired,
  fetchItems: PropTypes.func,
  noMoreFetch: PropTypes.bool.isRequired,
};

export default ItemList;