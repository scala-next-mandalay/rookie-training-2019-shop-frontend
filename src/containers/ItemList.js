import { connect } from 'react-redux'
import ItemList from '../components/ItemList'
import { 
  fetchAllItems,
} from '../modules/items'

const _getItemsByCategory = (rows, categoryId) => {
  if (categoryId <= 0) {
    return rows
  }
  else {
    return rows.filter(t => t.category_id === categoryId)
  }
}

export default connect(
  (state) => ({
    items: _getItemsByCategory(state.items.rows, state.items.selectedCateogryId),
    noMoreFetch: state.items.noMoreFetch
  }),
  (dispatch) => ({
    fetchAllItems: () => dispatch(fetchAllItems()),
  })
)(ItemList)
