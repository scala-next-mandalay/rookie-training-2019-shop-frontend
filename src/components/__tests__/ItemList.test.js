import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import { act } from 'react-dom/test-utils';
import ItemList from '../ItemList';
/*global expect*/
/*global jest*/
/*global Event*/

let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});
const noMoreFetch=true;

const items = [{ id: 1, name: "ver" ,price:4000 ,image: 'pic1' },
            { id: 2, name: "mer" ,price:3000 ,image: 'pic2' },
            { id: 3, name: "neko" ,price:1000 ,image: 'pic3' }];

const fetchItems=jest.fn(()=>{
    
});
describe("ItemList component", () => {
  it('matches the snapshot', () => {
    const ItemListSnapshot = renderer.create(
      <Parent><ItemList 
      items={items}
      noMoreFetch={noMoreFetch}
      fetchItems={fetchItems}
     /> </Parent>
    ).toJSON();
    expect(ItemListSnapshot).toMatchSnapshot();
  });
});