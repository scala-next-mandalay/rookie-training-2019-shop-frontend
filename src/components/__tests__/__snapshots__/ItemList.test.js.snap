// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList component matches the snapshot 1`] = `
Array [
  <div>
    <div
      className="MuiBox-root MuiBox-root-2"
      justify="center"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-106 MuiPaper-rounded"
          >
            <div
              className="MuiCardMedia-root makeStyles-media-107"
              style={
                Object {
                  "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/pic1\\")",
                }
              }
              title="ver"
            />
            <div
              className="MuiCardContent-root makeStyles-cardContent-108"
            >
              <div
                className="MuiBox-root MuiBox-root-141 nameTxt"
              >
                Name 
                ver
              </div>
              <div
                className="MuiBox-root MuiBox-root-142 priceTxt"
              >
                <span>
                  Price
                </span>
                :Price
                4000
                 Ks
              </div>
            </div>
            <div
              className="MuiCardActions-root makeStyles-cardActions-109 MuiCardActions-spacing"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <div
                    className="MuiBox-root MuiBox-root-165"
                  >
                    <span>
                      Add Button
                    </span>
                  </div>
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0zm18.31 6l-2.76 5z"
                      fill="none"
                    />
                    <path
                      d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-106 MuiPaper-rounded"
          >
            <div
              className="MuiCardMedia-root makeStyles-media-107"
              style={
                Object {
                  "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/pic2\\")",
                }
              }
              title="mer"
            />
            <div
              className="MuiCardContent-root makeStyles-cardContent-108"
            >
              <div
                className="MuiBox-root MuiBox-root-175 nameTxt"
              >
                Name 
                mer
              </div>
              <div
                className="MuiBox-root MuiBox-root-176 priceTxt"
              >
                <span>
                  Price
                </span>
                :Price
                3000
                 Ks
              </div>
            </div>
            <div
              className="MuiCardActions-root makeStyles-cardActions-109 MuiCardActions-spacing"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <div
                    className="MuiBox-root MuiBox-root-177"
                  >
                    <span>
                      Add Button
                    </span>
                  </div>
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M0 0h24v24H0zm18.31 6l-2.76 5z"
                      fill="none"
                    />
                    <path
                      d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;
